airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py

```diff
@@ -39,27 +39,36 @@ def __call__(self, uri: str) -> Dict[str, Any]:
 
 
 def resolve_ref_links(obj: Any) -> Any:
-    """
-    Scan resolved schema and convert jsonref.JsonRef object to JSON serializable dict.
+    """Resolve jsonref.JsonRef objects in a JSON schema and convert to JSON serializable dict.
+
+    Args:
+        obj (Any): The JSON schema object with ref field resolved.
 
-    :param obj - jsonschema object with ref field resolved.
-    :return JSON serializable object with references without external dependencies.
+    Returns:
+        Any: JSON serializable object with references converted and without external dependencies.
     """
-    if isinstance(obj, jsonref.JsonRef):
-        obj = resolve_ref_links(obj.__subject__)
-        # Omit existing definitions for external resource since
-        # we dont need it anymore.
-        if isinstance(obj, dict):
-            obj.pop("definitions", None)
-            return obj
+
+    def gen_items(container: Any) -> Generator:
+        if isinstance(container, dict):
+            stack.extend(container.items())
+            return container.items()
+        elif isinstance(container, list):
+            stack.extend(enumerate(container))
+            return enumerate(container)
+        else:
+            return []
+
+    stack = [obj]
+    while stack:
```

Comment:
I understand the stack-based approach is optimal, but the recursive implementation is easier to understand. This isn't part of the hot loop so I prefer keeping the existing implementation